linera-views/src/views/bucket_queue_view.rs

```diff
@@ -0,0 +1,700 @@
+        self.read_context(self.cursor.position, count).await
+    }
+
+    /// Returns the last element of a bucket queue view
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::context::create_test_memory_context;
+    /// # use linera_views::bucket_queue_view::BucketQueueView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_memory_context();
+    /// let mut queue = BucketQueueView::<_, u128, 5>::load(context).await.unwrap();
+    /// queue.push_back(34);
+    /// queue.push_back(37);
+    /// queue.push_back(47);
+    /// assert_eq!(queue.read_front(2).await.unwrap(), vec![34, 37]);
+    /// # })
+    /// ```
+    pub async fn read_back(&self, count: usize) -> Result<Vec<T>, ViewError> {
+        if count <= self.new_back_values.len() {
+            let start = self.new_back_values.len() - count;
+            Ok(self
+                .new_back_values
+                .range(start..)
+                .cloned()
+                .collect::<Vec<_>>())
+        } else {
+            let mut increment = self.count() - count;
+            let Some((i_block, mut position)) = self.cursor.position else {
+                unreachable!();
+            };
+            for block in i_block..self.stored_data.len() {
+                let size = self.stored_indices.indices[block].0 - position;
+                if increment < size {
+                    return self
+                        .read_context(Some((block, position + increment)), count)
+                        .await;
+                }
+                increment -= size;
+                position = 0;
+            }
+            unreachable!();
+        }
+    }
+
+    async fn load_all(&mut self) -> Result<(), ViewError> {
+        if !self.delete_storage_first {
+            let elements = self.elements().await?;
+            self.new_back_values.clear();
+            for elt in elements {
+                self.new_back_values.push_back(elt);
+            }
+            self.cursor = Cursor { position: None };
+            self.delete_storage_first = true;
+        }
+        Ok(())
+    }
+
+    /// Gets a mutable iterator on the entries of the queue
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::context::create_test_memory_context;
+    /// # use linera_views::bucket_queue_view::BucketQueueView;
+    /// # use linera_views::views::View;
+    /// # let context = create_test_memory_context();
+    /// let mut queue = BucketQueueView::<_, u8, 5>::load(context).await.unwrap();
+    /// queue.push_back(34);
+    /// let mut iter = queue.iter_mut().await.unwrap();
+    /// let value = iter.next().unwrap();
+    /// *value = 42;
+    /// assert_eq!(queue.elements().await.unwrap(), vec![42]);
+    /// # })
+    /// ```
+    pub async fn iter_mut(&'a mut self) -> Result<IterMut<'a, T>, ViewError> {
+        self.load_all().await?;
+        Ok(self.new_back_values.iter_mut())
+    }
```

Comment:
Yes, that is not great I agree. The problem is that the iterator is not async, so data has to be downloaded at once and then worked on.

Reply:
It could still be read all at once when creating the iterator.
But we could keep track of which values were actually _returned_ by the iterator: The ones that weren't can't have changed, so they wouldn't need to be rewritten when saving the view.

Reply:
I had this kind of issues for the `QueueView`. Yes, I think there is a possibility of improvement. However, I do not see it as critical.